### PR TITLE
Issue174

### LINKED ISSUE #174: For an offline K8s configuration, the Docker image being pulled every time although the Docker image is present on local machine
### Expected behaviour

For an offline K8s configuration, when applying basic.yaml, when necessary Docker images are present locally, it should not try to go online to pull the images at all.

### Actual behaviour

For an offline K8s cluster, we saved and loaded all necessary docker images with correct tags, etc. 

Despite this, each time a pod is being initialized, it immediately switches to Image Pull Back error state.

When we look at the logs using describe command, we can clearly see that it finds the necessary image locally but tries to pull the image from online Docker hub.

Note: When applying all-redis-operator-resources.yaml, we do not face this issue. (Docker image used for all-redis-operator-resources.yaml file is redis-operator:latest and for basic.yaml, it is redis:latest or redis:5-0-alpine)

### Environment

How are the pieces configured?
* Redis Operator version: v1
* Kubernetes version: v1.14

### PR DESCRIPTION
**Pull request by mcdiae**
Fixes #174

This fix allows the operator to be used on air-gapped clusters without access to an image registry.

Changes proposed on the PR:
- Expose `imagePullPolicy` for Redis, Sentinel and Exporter images
- Updates `custom-image.yaml` example spec

Note: the pull policy defaults to `Always` to keep existing behaviour. Code could be simplified if [default Kubernetes pull policies](https://kubernetes.io/docs/concepts/configuration/overview/#container-images) were used.

### CODE REVIEW

File: api/redisfailover/v1/types.go [hunk 1]
```diff
@@ -72,3 +76,20 @@ type RedisFailoverList struct {
 
 	Items []RedisFailover `json:"items"`
 }
+
+// ImagePullPolicy defines the pull policy with a default of Always
+type ImagePullPolicy corev1.PullPolicy
+
+// UnmarshalJSON sets the default value to Always on JSON decode
+func (e *ImagePullPolicy) UnmarshalJSON(b []byte) error {
```
Comment (chusAlvarez) [inline api/redisfailover/v1/types.go]:
```suggestion
func (i *ImagePullPolicy) UnmarshalJSON(b []byte) error {
```
Convention in go is use the first leter or leters of the type :)

File: api/redisfailover/v1/types.go [hunk 2]
```diff
@@ -72,3 +76,20 @@ type RedisFailoverList struct {
...
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+	if s == "" {
```
Comment (chusAlvarez) [inline api/redisfailover/v1/types.go]: You can simplify sintax:
```
	if s == "" {
		*e = ImagePullPolicy("Always")
	} else {
		*e = ImagePullPolicy(s)
	}
	return nil
```
to: 

```
	if s == "" {
		*e = ImagePullPolicy("Always")
		return nil
	} 

	*e = ImagePullPolicy(s)
	return nil
```

**Comment by chusAlvarez** [comment]
The patch is truly useful, could you add a check in generator_test.go for the new param :)?

**Comment by mcdiae** [comment]
Will clean up branch and resubmit

**Pull request closed without merging**